### ISSUE TITLE
lineage returns sql comments as part of node name

### ISSUE THREAD

**Issue by jaspev123** [tobymao/sqlglot#3414]
Having a snowflake query, sqlglot lineage parser returns nearby sql comments as part of node name - is this expected behaviour ?.
Below a test case and screen shot:

###  pip3 install sqlglot
import sys
import os

root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(root_dir)
current_file_path = os.path.abspath(__file__)
current_filename = os.path.basename(current_file_path)
print("** Testfile:", current_filename, '**')

import unittest
import sqlglot
from sqlglot.lineage import lineage
from sqlglot import  exp

class Test_SQLGLOT_COMMENT_BUG(unittest.TestCase):
    sql_content = """        
        CREATE TRANSIENT TABLE
            AAA.temp_table_123 DATA_RETENTION_TIME_IN_DAYS = 0
        AS
        SELECT DISTINCT
            *
        FROM (
            SELECT
                -- NO PROBLEM WITH THIS SQL COMMENT --
                MD5(NVL(UPPER(TRIM(some_column_B)), '-1') || '~' || NVL(UPPER(TRIM(another_column_A)), '-1') || '~' || NVL(UPPER(TRIM(comment_error_here)), '-1') || '~' || NVL(UPPER(TRIM(id)), '-1')) AS no_error_id
                -- CUSTOM SQL END --
                , xx_yy_number
                , some_ID_FIELD
                , some_column_B
                , another_column_A
                , comment_error_here
                -- ERROR SQL COMMENT START ERROR HERE HERE HERE HERE !!!! ---------------------------------------------------------------
                , MD5(NVL(UPPER(TRIM(some_column_B)), '-1') || '~' || NVL(UPPER(TRIM(another_column_A)), '-1') || '~' || NVL(UPPER(TRIM(comment_error_here)), '-1') || '~' || NVL(UPPER(TRIM(id)), '-1')) AS some_hash
                -- ERROR SQL COMMENT END --
                , 1 AS status
                , extra_column
            FROM
                domain.some_table_x x_table
            WHERE
                some_ID_FIELD IN (10,11,12,13,14)
            QUALIFY
                xx_yy_number = MAX(xx_yy_number) OVER (PARTITION BY some_column_B, another_column_A, comment_error_here)
        ) src
        WHERE
            NOT EXISTS (
                SELECT
                    1
                FROM
                    AAA.abc_dfg exists_relation
                WHERE
                    exists_relation.no_error_id = src.no_error_id
                    AND exists_relation.status = 1
            );
    """

    def test_bug_demonstration(self):
        dialect = "snowflake"
        ast = sqlglot.parse_one(self.sql_content, read=dialect)
        sql = ast.sql()
        for select in ast.find_all(exp.Select):
            for projection in select.expressions:
                column_lineage = {}
                column_name = projection.output_name
                try:
                    column_lineage = lineage((column_name), sql, dialect=dialect, trim_selects=False)
                except Exception as exception:
                    print(f'column_name: <' + column_name + f'> not parseable by lineage(): {exception}')
                if hasattr(column_lineage, 'downstream'):
                    for node in column_lineage.downstream:
                        if hasattr(node, 'downstream'):
                            if len(node.downstream) > 0:
                                print("node name:","<",str(node.downstream[0].name),">")

if __name__ == '__main__':
    unittest.main()

![image](https://github.com/tobymao/sqlglot/assets/14150443/47564e0c-7895-4908-b474-745b9b8ade8a)

**Comment by georgesittas** [comment]
Appreciate the report @jaspev123, fixed in #3418.

For future issues, it would be really helpful if you trimmed down the code that reproduces them to only what's necessary.

**Comment by jaspev123** [comment]
Thanks for rapid fix - unfortunately the fix broke
this parsing

```
def test_new_problem_after_previous_fix(self):
	problem_sql = """
	COPY INTO staging.STG_SOME_data_JSON
	FROM ( SELECT
			1712815413772 as stg_run_id,
			'SOME_data' as stg_source,
			$1 as json_content
		FROM
		   '@staging.stg_datadw/data/SOME_data/datadw/SOME_data_JSON/2024/04/11/'
	) 
	FILES = ('2024-04-11T06:03:24.378Z-alarms_overview.json.gz')
	FILE_FORMAT = (type='JSON' COMPRESSION = 'AUTO' ENABLE_OCTAL = FALSE ALLOW_DUPLICATE = TRUE STRIP_OUTER_ARRAY = FALSE STRIP_NULL_VALUES = FALSE IGNORE_UTF8_ERRORS = FALSE)
	;
	"""
	dialect = "snowflake"
	problem_ast = sqlglot.parse_one(problem_sql, read=dialect)  ## parse fails

```	


sqlglot.errors.ParseError: Expected table name but got <Token token_type: <TokenType.L_PAREN: 'L_PAREN'>, text: '(', line: 3, col: 7, start: 59, end: 59, comments: []>. Line 3, Col: 7.
  
        COPY INTO staging.STG_SOME_data_JSON
        FROM ( SELECT
                1712815413772 as stg_run_id,
                'SOME_data' as stg_source,
